src/sortStudents.ts

```diff
@@ -1,16 +1,70 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverage(grades: number[]): number {
+  return grades.reduce((acc, item) => {
+    return acc + item;
+  }, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents: Student[] = [...students];
+
+  switch (sortBy) {
+    case 'name':
+    case 'surname':
+      copyStudents.sort((a, b) => {
+        return order === 'asc'
+          ? a[sortBy].localeCompare(b[sortBy])
+          : b[sortBy].localeCompare(a[sortBy]);
+      });
+      break;
+
+    case 'age':
+      copyStudents.sort((a, b) => {
+        return order === 'asc' ? a[sortBy] - b[sortBy] : b[sortBy] - a[sortBy];
+      });
+      break;
+
+    case 'married':
```

Comment:
Do not repate your self
```suggestion
    case 'age':
    case 'married':
```